universum/analyzers/uncrustify.py

```diff
@@ -1,200 +1,202 @@
-                            help="(optional) Python 2.7 regular expression filter to apply to "
-                                 "combined list of files to check")
-        parser.add_argument("--output-directory", "-od", dest="output_directory", default="uncrustify",
-                            help="Directory to store fixed files, generated by Uncrustify "
-                                 "and HTML files with diff; the default value is 'uncrustify'")
-        utils.add_result_file_argument(parser)
-        return parser
-
-    def __init__(self, settings):
-        self.settings = settings
-        self.wrapcolumn = None
-        self.tabsize = None
-
-    def parse_files(self):
-        files = []
-        file_lines = []
-        for file_name in self.settings.file_names:
-            files.extend(add_files_recursively(file_name))
-        for file_list in self.settings.file_list_config:
-            with open(file_list) as f:
-                for file_name in f.readlines():
-                    file_lines.append(file_name.strip())
-        for file_name in file_lines:
-            files.extend(add_files_recursively(file_name))
-        for pattern in self.settings.pattern_form:
-            regexp = re.compile(pattern)
-            files = [file_name for file_name in files if regexp.match(file_name)]
-        files = [os.path.relpath(file_name) for file_name in files]
-        if not files:
-            sys.stderr.write("Please provide at least one file for analysis")
-            return 2
-
-        return files
-
-    def get_htmldiff_parameters(self):
-        if not (self.wrapcolumn and self.tabsize):
-            with open(self.settings.cfg_file) as config:
-                for line in config.readlines():
-                    if line.startswith("code_width"):
-                        self.wrapcolumn = int(line.split()[2])
-                    if line.startswith("input_tab_size"):
-                        self.tabsize = int(line.split()[2])
-        return self.wrapcolumn, self.tabsize
-
-    def generate_html_diff(self, file_name, left_lines, right_lines):
-        wrapcolumn, tabsize = self.get_htmldiff_parameters()
-        differ = difflib.HtmlDiff(tabsize=tabsize, wrapcolumn=wrapcolumn)
-        file_name = os.path.relpath(file_name, self.settings.output_directory).replace('/', '_') + '.html'
-        with open(os.path.join(self.settings.output_directory, file_name), 'w') as outfile:
-            outfile.write(differ.make_file(left_lines, right_lines, context=False))
-
-    def get_file_issues(self, src_file):
-        self.settings.output_directory = os.path.join(os.getcwd(), self.settings.output_directory)
-        # Uncrustify copies absolute path in its target folder, that's why we use '+'
-        uncrustify_file = os.path.normpath(self.settings.output_directory + '/' + src_file)
-
+def _uncrustify_output_parser(files: List[Tuple[Path, Path]],
```

Comment:
uncharacteristically, this analyzer ignores std output and instead uses result files to determine the issues - hence "_" parameter